app/src/main/java/de/tap/easy_xkcd/database/RealmComic.java

```diff
@@ -245,8 +245,16 @@ public static void saveOfflineBitmap(Response response, PrefHelper prefHelper, i
             if (!dir.exists()) {
                 dir.mkdirs();
             }
-            try (FileOutputStream fos = new FileOutputStream(sdCard.getAbsolutePath() + RealmComic.OFFLINE_PATH + "/" + comicFileName)) {
-                fos.write(response.body().bytes());
+
+            String comicFilePath = sdCard.getAbsolutePath() + RealmComic.OFFLINE_PATH + "/" + comicFileName;
+            File comicFile = new File(comicFilePath);
+
+            if (!comicFile.exists() || prefHelper.replaceOffline()) {
```

Comment:
Are you sure this is the right place to do it? Because at this point we already have downloaded the image anyway, so you won't save much time here. Instead, I think you should already check before downloading.

Reply:
I wasn't entirely sure _where_ the downloading happens. I'll see where the root of the tree is and move it there.